app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1684,6 +1685,17 @@ class PluginApi {
     addTopicTitleDecorator(callback);
   }
 
+  /**
+   * Allows a different limit to be set for fetching recent notifications for the user menu
+   *
+   * Example setting limit to 5:
+   * api.setUserMenuNotificationsLimit(5);
+   *
+   **/
+  setUserMenuNotificationsLimit(limit) {
```

Comment:
Is there a reason we went with a plugin API approach instead of a site setting? One problem with the plugin API approach here is that we are allowing any values to be set for the limit. However, there is actually an upper bound enforced by the server side API but the API does not communicate that to the user. 

If we need the number of initial items to be customisation, I feel like customising that via a site setting will be a better approach since we can validate the values of site settings.

Reply:
IMO It's nice to have a JS API for this so that developers can access this via themes without needing to rely on one of us changing a hidden site setting for them, or creating _yet another site setting in the admin UI_. I also don't love how styling/appearance is a bit mixed between themes and settings in your proposal. 

> Want to change the notifications panel? Style it in your theme, and then change a site setting! 

I prefer having a nice JS API that the theme can use.

___

This is actually necessary for a client because it's a feature flag, so the API is only used for certain users. But I wanted to make a case for it as a good idea in general, not just for client work.

Reply:
> there is actually an upper bound enforced by the server side API but the API does not communicate that to the user

Perhaps we could make the JS API throw an error if the value is higher than the hard-coded server-side limit?

Reply:
Good idea, good idea.